executor/distsql.go

```diff
@@ -646,23 +642,23 @@ func (w *indexWorker) fetchHandles(ctx context.Context, result distsql.SelectRes
 
 func (w *indexWorker) extractTaskHandles(ctx context.Context, chk *chunk.Chunk, idxResult distsql.SelectResult) (handles []int64, err error) {
 	handles = make([]int64, 0, w.batchSize)
-	for len(handles) < w.batchSize {
+	for {
 		err = errors.Trace(idxResult.Next(ctx, chk))
 		if err != nil {
 			return handles, err
 		}
 		if chk.NumRows() == 0 {
 			return handles, nil
 		}
-		for i := 0; i < chk.NumRows(); i++ {
-			handles = append(handles, chk.GetRow(i).GetInt64(0))
+		handles = append(handles, chunk.GetInt64InColumn(chk, 0)...)
+		if len(handles) >= w.batchSize {
+			if w.batchSize < w.maxBatchSize {
+				w.batchSize *= 2
+			}
+			return handles, nil
 		}
+		chk.Reset() // reset ahead so result.Next never grow chunk until `handles` reach batchSize
```

Comment:
1. Why not let `SelectResult` grow this `chk` in the next function call of `SelectResult.Next()`?
2. https://github.com/pingcap/tidb/pull/8903/files#diff-7492d01b7aac81533194642850a45b8eR118 it will call `GrowAndReset()`, seems this comment is incorrect.